### ISSUE TITLE
SMO issue with default column values.

### ISSUE THREAD

**Issue by sdonovanuk** [microsoft/sqlmanagementobjects#175]
Greetings. To reproduce. . . 

Create the following table:

```
CREATE TABLE DefaultTest (
    [FlagValue] INT DEFAULT 10
)
```

Then, generate the SMO script. You will get:

```
CREATE TABLE [dbo].[DefaultTest] (
    [FlagValue] INT DEFAULT ((10)) NULL
);
```

This is odd, why has the 10 become ((10))? Syntax wise, 10 is OK.
The same happens if you create the table without the default, then add it with:

` ALTER TABLE DefaultTest ADD DEFAULT 10 FOR [FlagValue]
`
What's interesting, is that if you use Azure Data Studio to edit the ((10)) default value in the UI back to 10, the SQL it generates displays as:

```
CREATE TABLE [dbo].[DefaultTest] (
    [FlagValue] INT DEFAULT 10 NULL
);
```

. . . which instantly becomes ((10)) again when you apply the change to the database.

Now, the engine always stores the constraints definition wrapped with ( and ). 
I contend there is a bug in SMO -- if the value within parenthesis is a primitive (string, integer, etc), then it doesn't need parenthesis.

Why is this a problem? Currently writing code to compare schemas against two databases, and having to parse weird output from DacFx's `ScriptComparison`, which is complaining that default values of (0) sometimes become ((0)). Now, I can write an AST parser using `ScriptDom`, but isn't this an SMO error?

Thanks!

**Comment by shueybubbles** [comment]
I 'd be reluctant to try to make SMO smart. Today it just gets the text of the `definition` and emits it. Doing anything else introduces a chance of emitting bad code.

**Comment by sdonovanuk** [comment]
"I'd be reluctant to try to make SMO smart". I can understand. However, it means that DacFx's `SchemaComparison` generates false-positives that you need to parse out, by writing your own (increasingly complex) derived object from `TSqlFragmentVisitor`. This negates the whole purpose of that DacFx offering schema comparator. This is clearly a bug, either SMO needs to tackle it, or `SchemaComparison` does. I'm concerned that SMO isn't generating technically identical scripts.

**Comment by llali** [comment]
@sdonovanuk I agree with you on schema compare needs lots of improvements, so it doesn't show differences on things that are basically the same thing. we are aware of some of those bugs but not all. if you please add any schema compare bug you can think of in dacfx repo (https://github.com/microsoft/DacFx/issues?q=is%3Aissue%20state%3Aopen) and we prioritize them in our future planning. However, schema compare doesn't use SMO for parsing/generating scripts so I'm not sure how this SMO bug can have any impact of schema compare.
